### PR TITLE
Add embedding integration with Intel Gaudi in llama-index-embeddings-gaudi

### PR DESCRIPTION
**Pull request by jeanyu-habana**
# Description
This PR adds Intel Gaudi support to the list of llama_index embeddings integration. This enables users to run embedding models locally on Intel Gaudi using llama_index modules.

Fixes # (issue)
N/A
## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [X] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [X] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                or self.embedding_input_size > self.max_seq_length
+            )
+            else self.embedding_input_size,
+            padding="max_length",
+            return_tensors="pt",
+            truncation=True,
+        )
+
+
+class GaudiHuggingFaceEmbeddings(HuggingFaceEmbeddings):
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py:41]: This is kind of un-intuitive -- can we implement the actual embedding class from llama-index right? No need to involve langhcain (You could subclass BaseEmbedding or the HuggingFaceEmbedding class from llama-index_

Comment (jeanyu-habana) [reply llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py]: Okay, I have just re-written the implementation to be based on BaseEmbedding of llama-index. @logan-markewich

**Comment by jeanyu-habana** [comment]
@logan-markewich Could you please review my updates when you have a chance? I'd appreciate it if the PR can be merged after issues are resolved. Thanks so much!

File: llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py]: Thank you!

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**